feature_engine/selection/drop_correlated_features.py

```diff
@@ -177,52 +187,35 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None):
             _check_contains_na(X, self.variables_)
             _check_contains_inf(X, self.variables_)
 
-        # set to collect features that are correlated
-        self.features_to_drop_ = set()
-
-        # create tuples of correlated feature groups
-        self.correlated_feature_sets_ = []
+        # sort features alphabetically to make selector deterministic
+        features = sorted(self.variables_)
 
         # the correlation matrix
-        _correlated_matrix = X[self.variables_].corr(method=self.method)
-
-        # create set of examined features, helps to determine feature combinations
-        # to evaluate below
-        _examined_features = set()
-
-        # for each feature in the dataset (columns of the correlation matrix)
-        for feature in _correlated_matrix.columns:
-
-            if feature not in _examined_features:
-
-                # append so we can exclude when we create the combinations
-                _examined_features.add(feature)
+        correlated_matrix = X[features].corr(method=self.method).to_numpy()
 
-                # here we collect potentially correlated features
-                # we need this for the correlated groups sets
-                _temp_set = set([feature])
-
-                # features that have not been examined, are not currently examined and
-                # were not found correlated
-                _features_to_compare = [
-                    f for f in _correlated_matrix.columns if f not in _examined_features
-                ]
-
-                # create combinations:
-                for f2 in _features_to_compare:
-
-                    # if the correlation is higher than the threshold
-                    # we are interested in absolute correlation coefficient value
-                    if abs(_correlated_matrix.loc[f2, feature]) > self.threshold:
-
-                        # add feature (f2) to our correlated set
-                        self.features_to_drop_.add(f2)
-                        _temp_set.add(f2)
-                        _examined_features.add(f2)
+        # the correlated pairs
+        correlated_mask = (abs(correlated_matrix) > self.threshold) - np.eye(
```

Comment:
Yep, it works. Thanks. Would this be faster? or more memory efficient?

Reply:
It will take upper triangle of the square matrix without the main diagonal (the one with the 1's). Should reduce the number of feature combinations to iterate over